octodns/provider/azuredns.py

```diff
@@ -94,21 +105,23 @@ def __init__(self, resource_group, record, delete=False):
         self.log = logging.getLogger('AzureRecord')
 
         self.resource_group = resource_group
-        self.zone_name = record.zone.name[:len(record.zone.name) - 1]
+        self.zone_name = record.zone.name[:-1]
         self.relative_record_set_name = record.name or '@'
         self.record_type = record._type
+        self._record = record
+        self.traffic_manager = traffic_manager
 
         if delete:
             return
 
         # Refer to function docstring for key_name and class_name.
         key_name = '{}_records'.format(self.record_type).lower()
         if record._type == 'CNAME':
-            key_name = key_name[:len(key_name) - 1]
+            key_name = key_name[:-1]
         azure_class = self.TYPE_MAP[self.record_type]
 
-        self.params = getattr(self, '_params_for_{}'.format(record._type))
-        self.params = self.params(record.data, key_name, azure_class)
+        params_for = getattr(self, '_params_for_{}'.format(record._type))
+        self.params = params_for(record.data, key_name, azure_class)
```

Comment:
Thanks for cleaning these bits up. Not sure how they were missed in the original PR review.

Reply:
The caveats text is too long for the doc-string, I've instead linked to this PR suggesting the reader to read it over.